helper_funcs.inc.php

```diff
@@ -51,70 +53,141 @@ function pwg_privacy_reject_access ($msg) {
 	exit();
 }
 
-function pwg_privacy_serve_file ($path) {
-	global $conf;
+function pwg_privacy_get_mime_type($file_path) {
+    // retrieve the mime type for the file. This is based on code found on php.net
+	$arrayZips = array("application/zip", "application/x-zip", "application/x-zip-compressed");
 
-	$ext = get_extension($path);
-	$ext = strtolower($ext);
+	$arrayExtensions = array(".pptx", ".docx", ".dotx", ".xlsx");
 
-	if (!in_array($ext, $conf['file_ext'])) {
-		pwg_privacy_reject_access('File extension is not allowed');
+	$original_extension = (false === $pos = strrpos($file_path, '.')) ? '' : substr($file_path, $pos);
+
+	$finfo = new finfo(FILEINFO_MIME);
+
+	$type = $finfo->file($file_path);
+
+	if (in_array($type, $arrayZips) && in_array($original_extension, $arrayExtensions))
+	{
+		return $original_extension;
+	}
+	// strip the charset encoding part
+	$type = preg_replace('#^([a-zA-Z*]*/[a-zA-Z*]*).*$#','$1',$type);
```

Comment:
Won't it be better to use `FILEINFO_MIME_TYPE` so you won't have the encoding in the first place? 
Also won't having the encoding cause the previous checks to always be false as the mime arrays do not contain any encoding info ?

Reply:
FILEINFO_MIME_TYPE isn't always available depending on php version. I'm not well versed in php. I've taken such statements at face value.

Reply:
From PHP doc `FILEINFO_MIME_TYPE` is available since php 5.3.0 so we can use it without any  worries.
It seem like this makes this whole function obsolete if we don't need to hack around zip files